pyasn1/codec/ber/decoder.py

```diff
@@ -276,13 +285,16 @@ def valueDecoder(self, substrate, asn1Spec,
         if tagSet[0].tagFormat != tag.tagFormatSimple:
             raise error.PyAsn1Error('Simple tag format expected')
 
-        the_bytes = substrate.read(length)
+        for the_bytes in substrate.read(length):
+            if the_bytes is None:
+                yield
+
         if not the_bytes:
-            return self._createComponent(asn1Spec, tagSet, 0, **options)
+            yield self._createComponent(asn1Spec, tagSet, 0, **options)
```

Comment:
We might have to yield on insufficient read, for that reason we become a coroutine as well. So we have to yield everywhere else and our caller has to treat us as a generator coroutine.